EnhancedMediaPlayer/MediaPlayerViewController.swift

```diff
@@ -0,0 +1,77 @@
+//  MediaPlayerViewController.swift
+
+import SwiftUI
+
+class MediaPlayerViewController: UIHostingController<MediaPlayerView> {
+    private let playerViewHandler: MediaPlayerControllerRepresentable
+    private let viewModel: MediaPlayerView.ViewModel
+
+    init(mediaURL: URL) {
+        self.viewModel = .init()
+        self.playerViewHandler = .init(mediaURL: mediaURL)
+
+        super.init(rootView: .init(viewModel: viewModel))
+        
+        self.viewModel.playerViewHandler = playerViewHandler
+        self.viewModel.playerState = .playing
+
+        setupActions()
+        observePlayToEnd()
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        removeObserver()
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupActions() {
+        viewModel.onTapAction = { [weak self] control in
+            guard let self else { return }
+            
+            switch control {
+                case .play: self.play()
+                case .pause: self.pause()
+                case .replay: self.replay()
+                case .rewind: self.rewind()
+                case .forward: self.forward()
```

Comment:
These are all strong references creating a retain cycle. This way the `MediaPlayerViewController` is never deallocated from memory. But if we add a `weak self` now the control buttons on the player will stop working... We should fix the other comments first, and solve this later on

Reply:
Actually, this was resolved when she changed the actions to be handled by an enum, this is already using the `weak` keyword and the retain cycle was fixed. But if you test the app you'll see that it's not working =p and that's because [this comment](https://github.com/profusion/ios-enhanced-media-player/pull/1#discussion_r1229985448) needs to be resolved. We can't instantiate a ViewController in a View's body like that and simply access its `rootView`, we need to create a Representable for it instead